eureka2-client/src/main/java/com/netflix/eureka2/client/interest/BatchingRegistry.java

```diff
@@ -0,0 +1,42 @@
+package com.netflix.eureka2.client.interest;
+
+import com.netflix.eureka2.interests.ChangeNotification;
+import com.netflix.eureka2.interests.Interest;
+import com.netflix.eureka2.interests.StreamStateNotification.BufferingState;
+import rx.Observable;
+
+/**
+ * This class collects information about batching statuses of individual atomic interests.
+ * It is assumed that state manipulation methods are serialized.
+ *
+ * @author Tomasz Bak
+ */
+public interface BatchingRegistry<T> {
+
+    /**
+     * Change notifications stream to observe. Subsequent subscribe invocation unsubscribe the
+     * previous subscription, and connect a new one.
+     */
+    void subscribe(Observable<ChangeNotification<T>> changeNotifications);
```

Comment:
Can we change the name here? This is pretty confusing from an rx perspective as in rx we subscribe to observables but the API here inverts that.